backend/WP/Endpoints/PostsEndpoint.php

```diff
@@ -31,7 +31,8 @@ public function endpoint_callback( \WP_REST_Request $request ) {
 			$args['s'] = $search;
 		}
 
-		$query       = Posts::query( $args );
```

Comment:
@nelsonamaya82 why did you changed this? Posts is a singleton class

Reply:
Tecnically if we want to avoid any possible instantation with `new` and create a "real" singleton we need to override the __construct method with the init method approach

Reply:
@Fcog @arzola  I did this because of this PHPMD Warning: http://phpmd.org/rules/cleancode.html#staticaccess

How do you think guys we can solve it? @Fcog what do you think about @arzola proposal? Makes sense to me but I'm not sure how to implement it, please help.

Reply:
ok, the singleton issue that's referring @arzola can be fixed on another PR.

The issue here is that PHPMD is not allowing a Static call inside this method because it becomes harder to test. I'm not completely sure, how to solve it.

Maybe instead of having Posts::query( $args ) in a Class, we can move it to a global function.

Reply:
alright, I replaced the Posts singleton for a functional approach by adding a global function instead. 
https://github.com/wearenolte/lean-theme/pull/136/files#diff-bd83ac1d46e5ecf421aca77587a177ba